src/v/redpanda/admin_server.cc

```diff
@@ -4057,6 +4060,106 @@ admin_server::get_partition_state_handler(
+            "min_spins_per_scheduling_point",
+            cfg.min_spins_per_scheduling_point);
+          parse_int(
+            "max_spins_per_scheduling_point",
+            cfg.max_spins_per_scheduling_point);
+          parse_int(
+            "min_ms_per_scheduling_point", cfg.min_ms_per_scheduling_point);
+          parse_int(
+            "max_ms_per_scheduling_point", cfg.max_ms_per_scheduling_point);
+          if (
+            cfg.max_spins_per_scheduling_point.has_value()
+            != cfg.min_spins_per_scheduling_point.has_value()) {
+              throw ss::httpd::bad_param_exception(
+                "Expected 'max_spins_per_scheduling_point' set with "
+                "'min_spins_per_scheduling_point'");
+          }
+          if (
+            cfg.max_ms_per_scheduling_point.has_value()
+            != cfg.min_ms_per_scheduling_point.has_value()) {
+              throw ss::httpd::bad_param_exception(
+                "Expected 'max_ms_per_scheduling_point' set with "
+                "'min_ms_per_scheduling_point'");
+          }
+          // Check that either delay or a spin count is defined.
+          if (
+            cfg.max_spins_per_scheduling_point.has_value()
+            == cfg.max_ms_per_scheduling_point.has_value()) {
+              throw ss::httpd::bad_param_exception(
+                "Expected either spins or delay to be defined");
+          }
+          if (
+            cfg.max_spins_per_scheduling_point.has_value()
+            && cfg.max_spins_per_scheduling_point.value()
+                 < cfg.min_spins_per_scheduling_point.value()) {
+              throw ss::httpd::bad_param_exception(fmt::format(
+                "Invalid parameter 'max_spins_per_scheduling_point' value "
+                "is too low: {} < {}",
+                cfg.max_spins_per_scheduling_point.value(),
+                cfg.min_spins_per_scheduling_point.value()));
+          }
+          if (
+            cfg.max_ms_per_scheduling_point.has_value()
+            && cfg.max_ms_per_scheduling_point.value()
+                 < cfg.min_ms_per_scheduling_point.value()) {
+              throw ss::httpd::bad_param_exception(fmt::format(
+                "Invalid parameter 'max_ms_per_scheduling_point' value "
+                "is too low: {} < {}",
+                cfg.max_ms_per_scheduling_point.value(),
+                cfg.min_ms_per_scheduling_point.value()));
+          }
+          cfg.num_fibers = 1;
+          if (auto e = req->get_query_param("num_fibers"); !e.empty()) {
+              try {
+                  cfg.num_fibers = boost::lexical_cast<int>(e);
+              } catch (const boost::bad_lexical_cast&) {
+                  throw ss::httpd::bad_param_exception(fmt::format(
+                    "Invalid parameter 'num_fibers' value {{{}}}", e));
+              }
+          }
+          return ss::smp::invoke_on_all([cfg, this] {
+                     auto ran = _stress_fiber_manager.local().start(cfg);
```

Comment:
should we support an auto stop feature (e.g. auto stop after N seconds) like we do for turning on trace logging?

Reply:
That's a good idea. I'll defer to a follow-up